Resources/Prototypes/Reagents/narcotics.yml

```diff
@@ -0,0 +1,119 @@
+﻿- type: reagent
+  id: Meth
+  name: meth
+  desc: Methamphetamine, more commonly know as meth, is a potent stimulant, with dangerous side-effects if too much is consumed.
+  physicalDesc: translucent
+  color: "#FAFAFA"
+  boilingPoint: 212.0 #Meth vape when?
+  meltingPoint: 170.0
+  metabolisms:
+    Poison:
+      effects:
+      - !type:HealthChange
+        damage:
+          types:
+            Poison: 1.5
+      - !type:HealthChange
+        conditions:
+        - !type:ReagentThreshold
+          min: 30
+        damage:
+          types:
+            Poison: 2 # this is added to the base damage of the meth.
+            Asphyxiation: 2
+    Narcotic:
+      effects:
+      - !type:MovespeedModifier
+        walkSpeedModifier: 1.3
+        sprintSpeedModifier: 1.3
+      - !type:GenericStatusEffect
+        key: Stutter
+        component: StutteringAccent
+      - !type:Jitter
+      - !type:GenericStatusEffect
+        key: Stun
+        time: 1
+        type: Remove
+      - !type:GenericStatusEffect
+        key: KnokedDown
+        time: 1
+        type: Remove
+
+- type: reagent
+  id: Ephedrine
+  name: ephedrine
+  desc: Increases stun resistance and movement speed, giving you hand cramps. Overdose deals toxin damage and inhibits breathing.
+  physicalDesc: Bone white
+  color: "#D2FFFA"
+  boilingPoint: 255.0
+  meltingPoint: 36.0
+  metabolisms:
+    Narcotic:
+      effects:
+      - !type:MovespeedModifier
+        walkSpeedModifier: 1.2
+        sprintSpeedModifier: 1.2
+      - !type:HealthChange
+        conditions:
+        - !type:ReagentThreshold
+          min: 20
+        damage:
+          types:
+            Poison: 2 # this is added to the base damage of the meth.
+            Asphyxiation: 2
+      - !type:Jitter
+      - !type:GenericStatusEffect
+        key: Stun
+        time: 0.25
+        type: Remove
+      - !type:GenericStatusEffect
+        key: KnockedDown
+        time: 0.25
+        type: Remove
+      - !type:PopupMessage
+        messages: ["ephedrine-effect-tight-pain", "ephedrine-effect-heart-pounds"]
+        type: Local
+        probability: 0.05
```

Comment:
I think the effect randomization works well enough, but could still be improved. For example while testing this drug I got no popups after using 30u, and then after trying again I got two identical pop-ups immediately after each other. I think it would be better if there were a way to have it be random, but also occur more consistently with a (randomized) minimum time interval. But I have no idea how you'd do that with how `ShouldApply` works atm. I guess that can just be done sometime in the future, if it's even worth addressing.

Reply:
a plan i had for this was to be able to apply reagent effects when a reagent enters your system and when it exits your system

Reply:
but that is out of scope, will do it later